sklearn/gaussian_process/gpr.py

```diff
@@ -0,0 +1,173 @@
+"""Gaussian processes regression. """
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+
+from sklearn.base import BaseEstimator, clone
+from sklearn.gaussian_process.kernels import RBF
+from sklearn.utils import check_random_state
+from sklearn.utils.validation import check_X_y, check_array
+
+
+class GaussianProcessRegressor(BaseEstimator):
+    """ Gaussian process regression (GPR).
+
+    The implementation is based on Algorithm 2.1 of ``Gaussian Processes
+    for Machine Learning'' (GPML) by Rasmussen and Williams.
+
+    In addition to standard sklearn estimators, GaussianProcessRegressor
+       * allows prediction without prior fitting (based on the GP prior)
+       * provides an additional method sample(X), which evaluates samples drawn
+         from the GPR (prior or posterior) at given inputs
+       * exposes a method log_marginal_likelihood(theta), which can be used
+         externally for other ways of selecting hyperparamters, e.g., via
+         Markov chain Monte Carlo.
+
+    Parameters
+    ----------
+    kernel : Kernel object
+        The kernel specifying the covariance function of the GP.
+
+    y_err : float, optional (default: 1e-10)
+        Value added to the diagonal of the kernel matrix during fitting.
+        Larger values correspond to increased noise level in the observations
+        and reduce potential numerical issue during fitting.
+
+    optimizer : string, optional (default: "fmin_l_bfgs_b")
+        A string specifying the optimization algorithm used for optimizing the
+        kernel's parameters. Default uses 'fmin_l_bfgs_b' algorithm from
+        scipy.optimize. If None, the kernel's paramters are kept fixed.
+        Available optimizers are::
+
+            'fmin_l_bfgs_b'
+
+    Attributes
+    ----------
+    X_fit_:
+
+    y_fit_:
+
+    theta_:
+
+    L_:
+
+    alpha_:
+    """
+
+    def __init__(self, kernel=None, y_err=1e-10, optimizer="fmin_l_bfgs_b"):
```

Comment:
`solver` sounds like it will identify the globally optimal solution, in my opinion. That makes sense for convex optimization problems but here we have a potentially multi-modal problem and there are no guarantees which local optimum is selected.  Thus, I would personally not call it `solver` here